dali/operators/reader/loader/tar_utils_test.cc

```diff
@@ -0,0 +1,215 @@
+  }
+};
+
+TEST_P(SimpleTarTests, Index) {
+  TestArchiveEntries(archive, GetParam().prefixes, GetParam().beg, GetParam().end, false);
+}
+
+TEST_P(SimpleTarTests, PostIndex) {
+  TestArchiveEntries(archive, GetParam().prefixes, GetParam().beg, GetParam().end, true);
+}
+
+TEST_P(SimpleTarTests, Contents) {
+  TAR* handle;
+  ASSERT_EQ(tar_open(&handle, GetParam().filepath.c_str(), NULL, O_RDONLY, 0, TAR_GNU), 0);
+
+  do {
+    th_read(handle);
+    vector<uint8_t> libtar_contents;
+    uint64_t count = 0;
+    uint64_t filesize = th_get_size(handle);
+
+    do {
+      libtar_contents.resize(libtar_contents.size() + T_BLOCKSIZE);
+      tar_block_read(handle, libtar_contents.data() + libtar_contents.size() - T_BLOCKSIZE);
+    } while ((count += T_BLOCKSIZE) < filesize);
+    libtar_contents.resize(filesize);
+
+    vector<uint8_t> file(archive.GetFileSize());
+    archive.Read(file.data(), archive.GetFileSize());
+
+    for (size_t idx = 0; idx < libtar_contents.size(); idx++) {
+      ASSERT_EQ(file[idx], libtar_contents[idx]);
+    }
+  } while (archive.NextFile());
+  ASSERT_EQ(th_read(handle), 1);
+  ASSERT_EQ(tar_close(handle), 0);
+}
+
+auto SimpleTarTestsValues() {
+  vector<SimpleTarTestsData> values;
+
+  SimpleTarTestsData filepaths[] = {
+      { dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                    "db/webdataset/MNIST/devel-0.tar"),
+        false,
+        false,
+        2000,
+        3000,
+        {".cls", ".jpg"} },
+      { dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                    "db/webdataset/sample-tar/empty.tar"),
+        false,
+        false,
+        0,
+        0,
+        {} },
+      { dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                    "db/webdataset/sample-tar/v7.tar"),
+        false,
+        false,
+        0,
+        1000,
+        {""} },
+      { dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                    "db/webdataset/sample-tar/oldgnu.tar"),
+        false,
+        false,
+        0,
+        1000,
+        {""} } };
+
+  for (auto& filepath : filepaths) {
+    for (int read_ahead = 0; read_ahead <= 1; read_ahead++) {
+      for (int use_mmap = 0; use_mmap <= 1; use_mmap++) {
+        filepath.read_ahead = read_ahead;
+        filepath.use_mmap = use_mmap;
+        values.push_back(filepath);
+      }
+    }
+  }
+  return testing::ValuesIn(values.begin(), values.end());
+}
+
+INSTANTIATE_TEST_CASE_P(LibTarUtilsTestParametrized, SimpleTarTests, SimpleTarTestsValues());
+
+
+constexpr int kMultithreadedSamples = 3;
+
+class MultiTarTests : public ::testing::TestWithParam<bool> {
+ protected:
+  std::unique_ptr<TarArchive> archives[kMultithreadedSamples];
+  const std::pair<int, int> ranges[kMultithreadedSamples] = {{2000, 3000}, {0, 1000}, {1000, 2000}};
+  void SetUp() override {
```

Comment:
That's a nitpick, but this should rather be `final`

Reply:
Done